akka-http-core/src/main/scala/akka/http/impl/engine/server/ServerTerminator.scala

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.http.impl.engine.server
+
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.Done
+import akka.actor.{ ActorSystem, Terminated }
+import akka.annotation.{ DoNotInherit, InternalApi }
+import akka.dispatch.ExecutionContexts
+import akka.event.LoggingAdapter
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.Http.{ HttpConnectionTerminated, HttpServerTerminated, HttpTerminated }
+import akka.http.scaladsl.settings.{ RoutingSettings, ServerSettings }
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse, StatusCodes }
+import akka.stream.scaladsl.BidiFlow
+
+import scala.concurrent.duration._
+import akka.stream._
+import akka.stream.stage._
+import akka.util.PrettyDuration
+
+import scala.annotation.tailrec
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+import scala.concurrent.duration.{ Deadline, FiniteDuration }
+
+/**
+ * Used to start the termination process of an Akka HTTP server.
+ *
+ * TODO more details here?
+ */
+// "Hasta la vista, baby."
+@DoNotInherit
+abstract class ServerTerminator {
```

Comment:
this is in `impl` package, is it still public API? then perhaps in wrong package?
otherwise InternalApi instead of DoNotInherit

Reply:
Yes public API, I will move it